### ISSUE TITLE
Stuck compiler path for Visual Studio 2017 after migrating to Visual Studio 2019

### ISSUE THREAD

**Issue by alexmcgilvery** [microsoft/vscode-cmake-tools#858]
### Brief Issue Summary

Trying to use Visual C++ 2019 compiler with multiple projects after migrating to Visual Studio Community 2019 from Visual Studio Community 2017, but cmake tools keeps using the path to the old Visual C++ 2017 compiler which was uninstalled with Visual Studio Community 2017.

**Tried resolutions:**
- Un+reinstalling CMake Tools extension.
- Un+updating to newest CMake version.
- Deleting prior .vscode + build folders in projects.
- Removed CMAKE_C(XX)_COMPILER from VSCode settings.json.
- Removed the Visual C++ 2017 compilers from saved kits.
- Adding compiler path to Windows system environment variables under CC + CXX.

**What works:**
- Visual Studio Community 2019's CMake builds the projects fine.
- Using the kit 'Clang 8.0.0 for MSVC with Visual Studio Community 2019 (amd64) works as usual.
- CMakeTools 'scan for kits' finds the Visual Studio Community 2019 variants.
- On my laptop where I never installed Visual Studio 2017; I've had no issues with the same setup using Visual Studio 2019.

I'm guessing it has something to do with the Ninja generator, as if I open CMake-GUI browse to the project and choose a different generator (Visual Studio) it will build using CMakeTools and the 2019 kit (This is not ideal since I wish to use Ninja, which I use for my Linux builds).

The Visual Studio 2017 path is apparently also cached in some settings somewhere other than my project directories, however I'm not sure where and how to clear this cache, which is why I'm opening this issue.

### Expected:

- CMake Tools finds the Visual C++ 2019 compiler.
- Builds using the Visual C++ 2019 compiler with Ninja.

### Apparent Behavior:

- Selecting Visual Studio Community 2019 - amd64 kit for building.
- Yields errors as shown below.

### CMake Tools Log
```
[cmake] The C compiler identification is unknown
[cmake] The CXX compiler identification is unknown
[cmake] CMake Error at CMakeLists.txt:9 (project):
[cmake]   The CMAKE_C_COMPILER:
[cmake] 
[cmake]     C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x64/cl.exe
[cmake] 
[cmake]   is not a full path to an existing compiler tool.
[cmake] 
[cmake]   Tell CMake where to find the compiler by setting either the environment
[cmake]   variable "CC" or the CMake cache entry CMAKE_C_COMPILER to the full path to
[cmake]   the compiler, or to the compiler name if it is in the PATH.
[cmake] 
[cmake] 
[cmake] CMake Error at CMakeLists.txt:9 (project):
[cmake]   The CMAKE_CXX_COMPILER:
[cmake] 
[cmake]     C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x64/cl.exe
[cmake] 
[cmake]   is not a full path to an existing compiler tool.
[cmake] 
[cmake]   Tell CMake where to find the compiler by setting either the environment
[cmake]   variable "CXX" or the CMake cache entry CMAKE_CXX_COMPILER to the full path
[cmake]   to the compiler, or to the compiler name if it is in the PATH.
```
### Platform and Versions
- **Operating System**:  Windows 10 -1903
- **CMake Version**: 3.15.5
- **VSCode Version**: 1.39.2
- **CMake Tools Extension Version**: 1.1.3
- **Compiler/Toolchain**: Visual C++ 2019

**Comment by bobbrow** [comment]
If you set your kit to the 2019 compiler and then run the "Delete cached build settings and reconfigure" command, does it still look for the 2017 compiler?

**Comment by alexmcgilvery** [comment]
Well I started up VsCode and the problem is no longer persisting. 

Unfortunately I didn't keep a detailed log on my attempts to fix, but I believe the last thing I tried was removing the CMAKE_C(XX)_COMPILER from settings.json. I hadn't restarted VsCode when I checked to see if that fix so, a restart might have been needed.

I had tried the "Delete cached build settings and reconfigure" command before, and it didn't resolve the issue. I'd tried all the cmake commands I thought where relevant.

Thanks for the assistance and sorry for the unnecessary issue creation.
